express/scripts/gnav.js

```diff
@@ -20,6 +20,55 @@ import {
 // eslint-disable-next-line import/no-unresolved
 } from './scripts.js';
 
+async function checkRedirect(currentPathName, geoLookup) {
+  const pathNameSplit = currentPathName.split('/');
+  let redirect = null;
+  if (pathNameSplit && pathNameSplit[1] && pathNameSplit[1] !== geoLookup) {
+    let subPath = geoLookup.length > 0 ? `/${geoLookup}` : '';
+    let i = 1;
+    while (pathNameSplit[i] !== 'express' && i < pathNameSplit.length) {
+      i += 1;
+    }
+    for (; i < pathNameSplit.length; i += 1) {
+      subPath = `${subPath}/${pathNameSplit[i]}`;
+    }
+    redirect = `${window.location.protocol}//${window.location.host}${subPath}${window.location.search}`;
+  }
+  return redirect;
+}
+
+async function geoCheck(userGeo, userLocale, geoCheckForce) {
+  const isInternationalCookie = document.cookie.match(/^(.*;)?\s*international\s*=\s*[^;]+(.*)?$/);
+  let redirect = null;
+  // DON'T if international cookie is set, unless geocheck=force is passed explicitly
+  if (geoCheckForce || !isInternationalCookie) {
+    const resp = await fetch(`${window.location.protocol}//${window.location.host}/express/system/geo-map.json`);
+    const json = await (resp.json());
+    let geoLookup = null;
+    for (let i = 0; i < json.data.length && geoLookup == null; i += 1) {
```

Comment:
Didn't `eslint` complain about this?
```suggestion
    for (let i = 0; i < json.data.length && geoLookup === null; i += 1) {
```

Reply:
Not at all - 0 eslint complaints! But let's be fully compliant.